Marlin/src/HAL/HAL_ESP32/fastio_ESP32.h

```diff
@@ -61,8 +61,8 @@
 #define extDigitalRead(IO)      digitalRead(IO)
 #define extDigitalWrite(IO,V)   digitalWrite(IO,V)
 
-#define PWM_PIN(P)              true
-#define USEABLE_HARDWARE_PWM(P) PWM_PIN(P)
+#define PWM_PIN(P)              (P < 34)
+#define USEABLE_HARDWARE_PWM(P) (TEST(P, 7) && PWM_PIN(P))
```

Comment:
All values less than 34 have bit 7 clear.

Reply:
Not the case for I2S expander which currently uses `PIN + 128` for pin indeces.

Reply:
The `< 34` is there anyway, since some of the ESP32 native GPIOs are input only.

Reply:
Also, with `&&` we now have incorrect behaviour - if it is I2S expander (`TEST(P, 7)` being true) then it'll checkout as a `USABLE_HARDWARE_PWM` pin, which we don't want.

Reply:
Oops, yeah, I must be dyslexic. Thanks for adding the `!`.